lpeg/io_modules/decoders/logfmt.lua

```diff
@@ -0,0 +1,135 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+--[[
+# Logfmt Message Decoder Module
+
+## Decoder Configuration Table for input cfg file
+decoder_module = "decoders.logfmt"
+
+-- Allows you to remap parsed logfmt tags into another field
+logfmt_conf = {
+    -- override root fields using Fields fields
+    remap_hash_root = {
```

Comment:
Some data best practices:
- It is generally a bad idea to alter the schema of an incoming message, if the schema needs to be changed it should be changed at the source.  Altering the schema breaks user expectations e.g.,  if I logfmt `level` as one of my key value pairs I wouldn't expect to look for it in msg.Severity. 
- Heka message headers are best used for metadata; in general avoid promoting data from a known schema format
- Not preserving type information is a failing of logfmt itself but mapping it in the cfg will just be a world of maintenance pain. Since logfmt is just a bag of key/value pairs and any app can log anything they want and change it at any time I would recommend something a little more dynamic.  Use the grammar to infer the type (at least number/string but you could also go deeper and include boolean/integer (although not perfect as one value may be foo = 1 and the the next message may be foo = 1.1 giving you two different schemas for the same type of message).  So instead of a manual mapping; settings like infer_type = true, numeric_type = "auto|double|integer" would be more flexible